compiler/lookups.nim

```diff
@@ -340,22 +340,35 @@ proc addOverloadableSymAt*(c: PContext; scope: PScope, fn: PSym) =
 proc addInterfaceDecl*(c: PContext, sym: PSym) =
   # it adds the symbol to the interface if appropriate
   addDecl(c, sym)
-  addInterfaceDeclAux(c, sym)
+  if not c.currentScope.isShadowScope:
```

Comment:
These don't cover all the entry points, but it's also not clear what should happen. :(